src/pybind/mgr/zabbix/module.py

```diff
@@ -0,0 +1,209 @@
+"""
+Zabbix module for ceph-mgr
+
+Collect statistics from Ceph cluster and every X seconds send data to a Zabbix
+server using the zabbix_sender executable.
+"""
+import json
+import subprocess
+from Queue import Queue, Empty
+from mgr_module import MgrModule
+
+
+class ZabbixSender(object):
+    def __init__(self, sender, host, port, log):
+        self.sender = sender
+        self.host = host
+        self.port = port
+        self.log = log
+
+    def send(self, hostname, data):
+        if len(data) == 0:
+            return
+
+        try:
+            cmd = '{0} -z {1} -p {2} -s {3} -vv -i -'.format(self.sender,
+                                                             self.host,
+                                                             self.port,
+                                                             hostname)
+
+            proc = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                                    stdout=subprocess.PIPE)
+
+            for key, value in data.items():
+                proc.stdin.write('{0} ceph.{1} {2}\n'.format(hostname, key,
+                                                             str(value)))
+
+            stdout, stderr = proc.communicate()
+            if proc.returncode != 0:
+                raise RuntimeError('%s exited non-zero' % self.sender)
+
+            self.log.debug('Zabbix Sender: %s', stdout.rstrip())
+        except:
+            raise
+
+
+class Module(MgrModule):
+    run = False
+
+    ceph_health_mapping = {'HEALTH_OK': 0, 'HEALTH_WARN': 1, 'HEALTH_ERR': 2}
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self.q = Queue()
+
+    def get_localized_config(self, key):
```

Comment:
It looks like this method is already copy-pasted in two modules, lets move it up into mgr_module.py in this PR